src/uag.c

```diff
@@ -499,15 +504,38 @@ static void sip_trace_handler(bool tx, enum sip_transp tp,
 			      const struct sa *src, const struct sa *dst,
 			      const uint8_t *pkt, size_t len, void *arg)
 {
+	int h, m, s;
+	uint64_t ms;
+
+#ifdef WIN32
+	SYSTEMTIME st;
+	GetSystemTime(&st);
+	ms = st.wMilliseconds;
+	h  = st.wHour;
+	m  = st.wMinute;
+	s  = st.wSecond;
+#else
+	struct timespec ts;
+	struct tm* local;
+
+	(void)clock_gettime(CLOCK_REALTIME, &ts);
+	ms = ts.tv_nsec / 1000000;
+	local = localtime(&ts.tv_sec);
+	h = local->tm_hour;
+	m = local->tm_min;
+	s = local->tm_sec;
+#endif
+
```

Comment:
it would be nice to make this code portable and put it in re.

we have some similar functions in re/src/fmt/time.c

did you have a look at `fmt_gmtime`